app/controllers/schedules_controller.rb

```diff
@@ -30,20 +31,27 @@ def events
     @dates = @conference.start_date..@conference.end_date
 
     @events_schedules = @program.selected_event_schedules
+    @events_schedules = @events_schedules.select{ |e| e.event.public_users.exists?(current_user.id) } if @events_schedules && current_user && @favourites
```

Comment:
And I would also prefer to have a dedicated view for this. Or is it too much intersection?